internal/dcache/cluster_manager/cluster_manager.go

```diff
@@ -724,23 +728,34 @@ func (cmi *ClusterManager) safeCleanupMyRVs(myRVs []dcache.RawVolume) (bool, err
 		maxWait = max(maxWait, time.Duration(cm.GetCacheConfig().ClustermapEpoch*3)*time.Second)
 
 		//
-		// Check status of all our RVs in the clustermap.
+		// Check status of all the RVs in the clustermap.
 		//
-		myRVsFromClustermap := cm.GetMyRVs()
+		rvsFromClustermap := cm.GetAllRVs()
 
-		if len(myRVsFromClustermap) > 0 {
-			log.Info("ClusterManager::safeCleanupMyRVs: Got %d of my RV(s) from clustermap %+v",
-				len(myRVsFromClustermap), myRVsFromClustermap)
+		if len(rvsFromClustermap) > 0 {
+			log.Info("ClusterManager::safeCleanupMyRVs: Got %d of RV(s) from clustermap %+v",
+				len(rvsFromClustermap), rvsFromClustermap)
 		} else {
-			log.Info("ClusterManager::safeCleanupMyRVs: No my RV(s) in clustermap")
+			log.Info("ClusterManager::safeCleanupMyRVs: No RV(s) in clustermap")
+		}
+
+		// Ensure no two different cache-dir report the same rvId(filesystem GUID)
+		for _, inputRV := range myRVs {
+			for _, cmRV := range rvsFromClustermap {
+				if inputRV.RvId == cmRV.RvId && (inputRV.LocalCachePath != cmRV.LocalCachePath || cmRV.NodeId != cmi.myNodeId) {
+					return false, fmt.Errorf(
+						"ClusterManager::safeCleanupMyRVs: Duplicate RVid (filesystem GUID) %s detected. Input path/node %s/%s conflicts with existing path/node %s/%s",
+						inputRV.RvId, inputRV.LocalCachePath, cmi.myNodeId, cmRV.LocalCachePath, cmRV.NodeId)
+				}
+			}
 		}
 
 		rvStillOnline := false
 		for _, rv := range myRVs {
 			log.Info("ClusterManager::safeCleanupMyRVs: Checking my RV %+v", rv)
 
 			// Check online status for this RV.
-			for rvName, rvInfo := range myRVsFromClustermap {
+			for rvName, rvInfo := range rvsFromClustermap {
```

Comment:
we want to do this only for our RVs and not for all RVs.
Only the check for dup rvid, we need to do for all RVs, the offline check we need to do only for our RVs as that's all we care about.